java/internalmetrics.md

```diff
@@ -0,0 +1,185 @@
+|:---|:---:|:---|
+|~sdk.java.core.sender.proxy.spans.discarded.count                      |Counter    |Spans discarded due to port(s) that were not configured.|
+|~sdk.java.core.sender.proxy.spans.valid.count                          |Counter    |Valid spans received.|
+|~sdk.java.core.sender.proxy.spans.invalid.count                        |Counter    |Invalid spans received.|
+|~sdk.java.core.sender.proxy.spans.dropped.count                        |Counter    |Spans dropped due to failure to write to socket.|
+|~sdk.java.core.sender.proxy.tracingHandler.errors.count                |Counter    |Errors encountered, excluding connection errors. |
+|~sdk.java.core.sender.proxy.tracingHandler.connect.errors.count        |Counter    |Errors encountered connecting to remote socket.|
+|~sdk.java.core.sender.proxy.tracingHandler.socket.write.success.count  |Counter    |Successful writes to socket.|
+|~sdk.java.core.sender.proxy.tracingHandler.socket.write.errors.count   |Counter    |Unsuccessful writes to socket.|
+|~sdk.java.core.sender.proxy.tracingHandler.socket.flush.success.count  |Counter    |Successful socket flushes.|
+|~sdk.java.core.sender.proxy.tracingHandler.socket.flush.errors.count   |Counter    |Unsuccessful socket flushes.|
+|~sdk.java.core.sender.proxy.tracingHandler.socket.reset.success.count  |Counter    |Successful connection resets.|
+|~sdk.java.core.sender.proxy.tracingHandler.socket.reset.errors.count   |Counter    |Unsuccessful connection resets.|
+
+## Metrics Collected When Using Direct Ingestion
+If you directly sending data to Wavefront using a `WavefrontDirectIngestionClient` instance, the following diagnostic metrics are reported:
+
+* [Metric Points Ingestion](#Metric-Points-Ingestion)
+* [Histogram Distributions Ingestion](#Histogram-Distributions-Ingestion)
+* [Tracing Spans Ingestion](#Tracing-Spans-Ingestion)
+
+### Metric Points Ingestion
+|Metric Name|Metric Type|Description|
+|:---|:---:|:---|
+|~sdk.java.core.sender.direct.points.queue.size                 |Gauge      |Points in ingestion buffer.|
+|~sdk.java.core.sender.direct.points.queue.remaining_capacity   |Gauge      |Remaining capacity of ingestion buffer.|
+|~sdk.java.core.sender.direct.points.valid.count                |Counter    |Valid points received.|
+|~sdk.java.core.sender.direct.points.invalid.count              |Counter    |Invalid points received.|
+|~sdk.java.core.sender.direct.points.dropped.count              |Counter    |Valid points that are dropped during ingestion.|
+|~sdk.java.core.sender.direct.points.report.errors.count        |Counter    |Exceptions encountered while reporting points.|
+|~sdk.java.core.sender.direct.points.report.202.count           |Counter    |Report requests by response status code.|
+
+### Histogram Distributions Ingestion
+|Metric Name|Metric Type|Description|
+|:---|:---:|:---|
+|~sdk.java.core.sender.direct.histograms.queue.size                 |Gauge      |Distributions in ingestion buffer.|
```

Comment:
I'm puzzled that isn't a counter, but maybe I'm misunderstanding something.

Reply:
Checked with Han, and he confirmed the current content.